packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,963 @@
+      return;
+    }
+
+    this.displayInput.focus({ preventScroll: true });
+    this.open = !this.open;
+  }
+
+  private handleComboboxKeyDown(event: KeyboardEvent) {
+    if (event.key === 'Tab') {
+      return;
+    }
+
+    event.stopPropagation();
+    this.handleDocumentKeyDown(event);
+  }
+
+  private handleClearClick(event: MouseEvent) {
+    event.stopPropagation();
+    this.clearCombobox();
+    // TODO: clearify if the listbox should stay open like this or be closed
+  }
+
+  private clearCombobox() {
+    if (this.value !== '') {
+      this.value = '';
+      this.displayInput.value = '';
+      this.lastOptionValue = '';
+      this.setSelectedOption(undefined);
+      this.displayInput.focus({ preventScroll: true });
+
+      // Emit after update
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.updateComplete.then(() => {
+        this.emit('syn-clear');
+        this.emit('syn-input');
+        this.emit('syn-change');
+      });
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private preventLoosingFocus(event: MouseEvent) {
+    // Don't lose focus of the input or propagate events
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  /* eslint-disable @typescript-eslint/no-floating-promises */
+  private handleOptionClick(event: MouseEvent) {
+    const target = event.target as HTMLElement;
+    const option = target.closest('syn-option');
+    const oldValue = this.lastOptionValue;
+    if (option && !option.disabled) {
+      this.setSelectedOption(option);
+
+      // Set focus after updating so the value is announced by screen readers
+      this.updateComplete.then(() => this.displayInput.focus({ preventScroll: true }));
+
+      if (this.value !== oldValue) {
+        // Emit after updating
+        this.updateComplete.then(() => {
+          this.emit('syn-input');
+          this.emit('syn-change');
+        });
+      }
+
+      this.hide();
+      this.displayInput.focus({ preventScroll: true });
+    }
+  }
+  /* eslint-enable @typescript-eslint/no-floating-promises */
+
+  /**
+   * Selects the following or previous option.
+   *
+   * @param isNext - A boolean indicating whether to select the following option (true)
+   *                 or the previous option (false).
+   */
+  private selectNextOption(isNext: boolean) {
+    const filteredOptions = this.getAllFilteredOptions();
+    const currentIndex = this.currentOption ? filteredOptions.indexOf(this.currentOption) : -1;
+    let newIndex = Math.max(0, currentIndex);
+
+    if (isNext) {
+      newIndex = currentIndex + 1;
+      if (newIndex > filteredOptions.length - 1) newIndex = 0;
+    } else {
+      newIndex = currentIndex - 1;
+      if (newIndex < 0) newIndex = filteredOptions.length - 1;
+    }
+    this.setCurrentOption(filteredOptions[newIndex]);
+    scrollIntoView(this.currentOption!, this.listbox, 'vertical', 'auto');
+  }
+
+  private getAllFilteredOptions() {
```

Comment:
I am not sure where this comes from, but when using the demos, you will get duplicates for each item. In the demo applications, you have entries for `American` and `american`, but only `American` is set as an option initially. This happens in all framework demo projects. Mind to have a look into this?

Reply:
This can also be seen in the "Highlight Query" demo